prymer/primer3/primer3_parameters.py

```diff
@@ -204,6 +217,10 @@ def __post_init__(self) -> None:
             raise TypeError("Probe melting temperatures and GC content must be floats")
         if self.probe_max_dinuc_bases % 2 == 1:
             raise ValueError("Max threshold for dinucleotide bases must be an even number of bases")
+        default_thermo_max: float = self.probe_tms.min - 10.0
+        for field in fields(self):  # if *_thermo fields are None, set to default
+            if field.type is Optional[float] and getattr(self, field.name) is None:
+                object.__setattr__(self, field.name, default_thermo_max)
```

Comment:
**suggestion** I think we should explicitly check that a field is one of the `probe_max_*_thermo` fields before setting the max.

Not an issue right now but it's defensive against the introduction of other `Optional[float]` attributes

```suggestion
        thermo_max_fields = ["probe_max_self_any_thermo", "probe_max_self_any_thermo", "probe_max_hairpin_thermo"]
        for field in fields(self):
            if field.name in thermo_max_fields and getattr(self, field.name) is None:
                object.__setattr__(self, field.name, default_thermo_max)
```

(you could also do something like `field.name.startswith("probe_max_")` if that's sufficiently specific